test/sagas/session_test.js

```diff
@@ -37,21 +39,77 @@ describe("session sagas", () => {
       getState = () => ({
         session: sessionState,
       });
-      const action = actions.setup(authData);
-      setupSession = saga.setupSession(getState, action);
-      setupSession.next();
+      action = actions.getServerInfo(authData);
+      getServerInfo = saga.getServerInfo(getState, action);
+    });
 
-      client = getClient();
+    describe("Success", () => {
+      it("should call client.fetchServerInfo", () => {
+        const fetchServerInfoCall = getServerInfo.next().value;
+        client = getClient();
+        expect(fetchServerInfoCall).eql(call([client, client.fetchServerInfo]));
+      });
+
+      it("should have configured the client", () => {
+        expect(getClient().remote).eql(authData.server);
+      });
+
+      it("should send a serverInfoSuccess", () => {
+        expect(getServerInfo.next(serverInfo).value).eql(
+          put(actions.serverInfoSuccess(serverInfo))
+        );
+      });
+
+      it("should clear the notifications", () => {
+        expect(getServerInfo.next().value).eql(
+          put(notificationsActions.clearNotifications({ force: true }))
+        );
+      });
     });
 
-    it("should configure the client", () => {
-      expect(getClient().remote).eql(authData.server);
+    describe("Failure", () => {
+      it("should reset the server info", () => {
```

Comment:
just out of curiosity, why should we reset server info?

Reply:
If you enter a bogus server in the input box, you don't want to still have the capabilities from the previous "working" server. Does that answer your question? Did I understand your question correctly?

Reply:
yep! got it, wasn't obvious from the reviewer standpoint :)

Reply:
Good point, I'll add a comment in the code ;) Thanks for bringing it up!